### PR TITLE
Add OpenSSL 1.1 to Fedora 36 to support VMR Codespaces

### PR DESCRIPTION
**Pull request by premun**
https://github.com/dotnet/arcade/issues/12101

### CODE REVIEW

File: src/fedora/36/amd64/Dockerfile [hunk 1]
```diff
@@ -51,3 +51,9 @@ RUN dnf --setopt=install_weak_deps=False install -y \
         uuid-devel \
         zlib-devel \
     && dnf clean all
+
+# Dependencies for running the VMR tooling
+# LibGit2Sharp requires this when synchronizing the VMR in a Codespace
+RUN dnf --setopt=install_weak_deps=False install -y \
+        openssl1.1.x86_64 \
```
Comment (mthalman) [inline src/fedora/36/amd64/Dockerfile:58]: The `.x86_64` portion of this is implied based on the architecture of the container. I think it'd be better to omit this just for readability.

Comment (premun) [reply src/fedora/36/amd64/Dockerfile:58]: Thanks!

```suggestion
        openssl1.1 \
```

**Review by mthalman — APPROVED** [review]

**Comment by premun** [comment]
@mthalman do you know when this new image will be available in the registry?

**Comment by mthalman** [comment]
It's currently available from staging using the `mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-36-staging` tag. So you can use that for testing. It'll be rolled out to the production `mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-36` tag as part of weekly rollout done by @riarenas / @michellemcdaniel -- probably Wednesday?

**Comment by riarenas** [comment]
> probably Wednesday?

Yep, that is still the plan for this week.

**Comment by premun** [comment]
@mthalman I tried to use the new image but nor the production or the staging one seemed to have this package installed. Is there something I'm doing wrong?

I did this to double check:
```
docker run --rm -it mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-36-staging
[root@f88a33afef8e /]# dnf list installed | grep ssl
openssl-devel.x86_64                     1:3.0.5-2.fc36             @updates
openssl-libs.x86_64                      1:3.0.5-2.fc36             @koji-override-1
```

Then I tried:
```
[root@f88a33afef8e /]# dnf -y install openssl1.1
...
[root@f88a33afef8e /]# dnf list installed | grep ssl
openssl-devel.x86_64                     1:3.0.5-2.fc36             @updates
openssl-libs.x86_64                      1:3.0.5-2.fc36             @koji-override-1
openssl1.1.x86_64                        1:1.1.1q-1.fc36            @updates         <<<<< This is here now
```

Are there logs available from build of the container where I can verify this is being installed?

**Comment by mthalman** [comment]
Sorry about that. The publish pipeline was stuck for days waiting on a permission prompt. I've kicked it and the staging image is now updated. The production image should be updated later today.

**Comment by premun** [comment]
Thanks!

**Comment by premun** [comment]
Worked, thanks!

**Pull request merged**